src/test/fuzz/addrman.cpp

```diff
@@ -11,25 +11,37 @@
 #include <test/fuzz/FuzzedDataProvider.h>
 #include <test/fuzz/fuzz.h>
 #include <test/fuzz/util.h>
+#include <test/util/setup_common.h>
 #include <time.h>
 #include <util/asmap.h>
+#include <util/system.h>
 
 #include <cassert>
 #include <cstdint>
 #include <optional>
 #include <string>
 #include <vector>
 
+static std::unique_ptr<const BasicTestingSetup> g_testing_setup;
+
 void initialize_addrman()
 {
-    SelectParams(CBaseChainParams::REGTEST);
+    if (!g_testing_setup) {
```

Comment:
Why is this check needed?

Reply:
`initialize_addrman()` will be called from `data_stream_addr_man`, `addrman` and `addrman_serdeser` tests. Without this check it will create and destroy the `BasicTestingSetup` object before/after each test. I do not know if this is a problem. The code I mimic (and which presumably works) is using `static` variable inside the `initialize_...()` function:

https://github.com/bitcoin/bitcoin/blob/77a2f5d30c5ecb764b8a7c098492e1f5cdec90f0/src/test/fuzz/banman.cpp#L30-L32

https://github.com/bitcoin/bitcoin/blob/77a2f5d30c5ecb764b8a7c098492e1f5cdec90f0/src/test/fuzz/coins_view.cpp#L40-L42

https://github.com/bitcoin/bitcoin/blob/77a2f5d30c5ecb764b8a7c098492e1f5cdec90f0/src/test/fuzz/load_external_block_file.cpp#L20-L22

https://github.com/bitcoin/bitcoin/blob/77a2f5d30c5ecb764b8a7c098492e1f5cdec90f0/src/test/fuzz/net.cpp#L24-L26

These `static` variables will be initialized only once, even if the `initialize...()` function is called multiple times. So I mimic that with the `if` check.

Reply:
It is only possible to run one fuzz target at a time and it is not possible to switch the fuzz target.

The reasons to use the initialize function is because globals that are initialized before main are bad. For fuzzing that means even globals that are never used can poison the coverage data or crash the fuzz target. Also, using the init function allows the fuzz engine to properly set up its own accounting before and exactly know when the fuzz targets own initialization code happens.

Reply:
Why are those `initialize...()` functions using `static` variables?

Reply:
`static` inside a function will only call the constructor when called the first time.

They use `static` because the constructed object needs to live after the function returned.

Reply:
> They use `static` because the constructed object needs to live after the function returned

Oh, yes, of course. I completely missed that aspect. Was only focusing on the fact that the `static` variables will be initialized only once even if the function is called multiple times.

Changed now to do like the other tests, e.g. `src/test/fuzz/coins_view.cpp`.